src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,142 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public class Node<E> {
+        private Node<E> prev;
+        private E value;
+        private Node<E> next;
+
+        public Node(Node<E> prev, E value, Node<E> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (isEmpty()) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+        } else {
+            Node<T> current = getIndex(index);
+            Node<T> newNode = new Node<>(current.prev, value, current);
+            current.prev.next = newNode;
+            current.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T returnValue = getIndex(index).value;
+        getIndex(index).value = value;
+        return returnValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T returnValue = getIndex(index).value;
+        unlink(getIndex(index));
+        return returnValue;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        for (int i = 0; i < size; i++) {
+            if (isEqual(current, object)) {
```

Comment:
suggestion about parameters that make method isEqual() more readable:
```suggestion
            if (isEqual(current.value, object)) {
```